libdnf/dnf-repo.cpp

```diff
@@ -1418,57 +1425,28 @@ dnf_repo_check_internal(DnfRepo *repo,
         }
     }
 
+    g_hash_table_remove_all(priv->filenames_md);
+    for (auto *elem = yum_repo->paths; elem; elem = g_slist_next(elem)) {
+        if (elem->data) {
+            auto yumrepopath = static_cast<LrYumRepoPath *>(elem->data);
+            g_hash_table_insert(priv->filenames_md, g_strdup(yumrepopath->type),
+                                g_strdup(yumrepopath->path));
+        }
+    }
+
     /* create a HyRepo */
     if (priv->repo != NULL)
         hy_repo_free(priv->repo);
     priv->repo = hy_repo_create(priv->id);
     hy_repo_set_string(priv->repo, HY_REPO_MD_FN, yum_repo->repomd);
-    tmp = lr_yum_repo_path(yum_repo, "primary");
-    if (tmp != NULL) {
-        hy_repo_set_string(priv->repo, HY_REPO_PRIMARY_FN, tmp);
-        g_hash_table_insert(priv->filenames_md,
-                            g_strdup("primary"),
-                            g_strdup(tmp));
-    }
-    tmp = lr_yum_repo_path(yum_repo, "filelists");
-    if (tmp != NULL) {
-        hy_repo_set_string(priv->repo, HY_REPO_FILELISTS_FN, tmp);
-        g_hash_table_insert(priv->filenames_md,
-                            g_strdup("filelists"),
-                            g_strdup(tmp));
-    }
-    tmp = lr_yum_repo_path(yum_repo, "updateinfo");
-    if (tmp != NULL) {
-        hy_repo_set_string(priv->repo, HY_REPO_UPDATEINFO_FN, tmp);
-        g_hash_table_insert(priv->filenames_md,
-                            g_strdup("updateinfo"),
-                            g_strdup(tmp));
-    }
-    tmp = lr_yum_repo_path(yum_repo, "modules");
-    if (tmp != nullptr) {
-        hy_repo_set_string(priv->repo, MODULES_FN, tmp);
-        g_hash_table_insert(priv->filenames_md,
-                            g_strdup("modules"),
-                            g_strdup(tmp));
-    }
-    tmp = lr_yum_repo_path(yum_repo, "group");
-    if (tmp != NULL) {
-        g_hash_table_insert(priv->filenames_md,
-                            g_strdup("group"),
-                            g_strdup(tmp));
-    }
-    tmp = lr_yum_repo_path(yum_repo, "appstream");
-    if (tmp != NULL) {
-        g_hash_table_insert(priv->filenames_md,
-                            g_strdup("appstream"),
-                            g_strdup(tmp));
-    }
-    tmp = lr_yum_repo_path(yum_repo, "appstream-icons");
-    if (tmp != NULL) {
-        g_hash_table_insert(priv->filenames_md,
-                            g_strdup("appstream-icons"),
-                            g_strdup(tmp));
-    }
+    tmp = static_cast<const gchar *>(g_hash_table_lookup(priv->filenames_md, "primary"));
+    hy_repo_set_string(priv->repo, HY_REPO_PRIMARY_FN, tmp);
+    tmp = static_cast<const gchar *>(g_hash_table_lookup(priv->filenames_md, "filelists"));
```

Comment:
It is optional file. Is the case where file is not described in repomd covered?

Reply:
Yes, some files are optional and the new code covers these cases.